src/components/Results.tsx

```diff
@@ -0,0 +1,57 @@
+import { Component } from 'react';
+import { BookSearchResponse } from '../api/models';
+import { ErrorContext } from './ErrorBoundary';
+
+interface ResultsProps {
+  response?: BookSearchResponse;
+  isLoading: boolean;
+}
+
+export class Results extends Component<ResultsProps> {
+  static contextType = ErrorContext;
+  declare context: React.ContextType<typeof ErrorContext>;
+
+  render() {
+    const error = this.context?.error && (
+      <h2 className="text-base/7 font-semibold text-red-600">
+        ❌ {this.context.error.message}
+      </h2>
+    );
+
+    const loader = this.props.isLoading && (
+      <h2 className="animate-pulse text-base/7 font-semibold">Loading...</h2>
+    );
+
+    const books =
+      this.props.response &&
+      this.props.response?.docs.map((book, i) => {
```

Comment:
[minor]: if we check the `response` with AND operator, then we don't need optional chaining in the next line, or we can get rid of first check:
```suggestion
      this.props.response?.docs.map((book, i) => {
```

Reply:
refactored in [refactor: refactor Results](https://github.com/programister42/rs-react-app/pull/1/commits/6def239ca4e9a515ed028f9584ac4f39b0b3eefc)